src/main/kotlin/no/nav/familie/ef/sak/repository/domain/Vedtaksbrev.kt

```diff
@@ -0,0 +1,14 @@
+package no.nav.familie.ef.sak.repository.domain
+
+
+import no.nav.familie.ef.sak.api.dto.BrevRequest
+import no.nav.familie.ef.sak.service.steg.StegType
+import org.springframework.data.annotation.Id
+import java.util.*
+
+data class Vedtaksbrev(@Id
+                       val id: UUID = UUID.randomUUID(),
+                       val behandlingId: UUID,
+                       val steg: StegType,
+                       val brevRequest: BrevRequest,
```

Comment:
Hvorfør trenger man lagre brevRequestet ?

Reply:
Fordi vi skal generere brevet på nytt for totrinnskontroll. Da lagrer vi de parametrene saksbehandler 1 "konfigurerte" brevet med i `brevRequest` så vi kan lage identisk brev + ny signatur ved totrinnskontroll.

Reply:
Aha! Men borde vi ikke hente den datan fra befintlige tabeller og ikke bruke request. Eksempelvis belopOvergangsstonad borde hentes fra TilkjentYtelseTabellen ? Men detta er sikkert dere tänkt på

Reply:
Men det er ikke all den dataen som finnes i andre tabeller :) Det kommer til å være en del brev-spesifikke parametre som saksbehandler 1 fyller inn -  typ hvilke tekster som skal brukes osv.

Reply:
Dataen som finnes i andre tabeller som er relevant for brevet er jeg usikker på om bør lagres i `brevRequest` eller ikke. Fordelen med å lagre "all" data i `brevRequest`-objektet er at vi kan sende den, uforandret til `familie-brev`-tjenesten, og få det samme brev hver gang. Hvis vi skal bygge opp en request til brevtjenesten ved å hente data fra flere tabeller frykter jeg at vi ikke har 100% på kontroll på at saksbehandler 2 ser nøyaktig samme brev som saksbehandler 1. Men, vi kan vel egentlig forvente at alle tabeller er "stabile" hvis vi implementerer ting riktig.

Reply:
Skjönner! Jag frykter att hvis man mellomlagrer brevRequesten og ikke bruker befintlig data i tabellerne att det finns en riseko for missmatch mellom vad som står i brevet og hva som faktiskt blev utfallet. Er planen att man ska "låse" behandligen for redigering når man er på steget "SEND_TIL_BESLUTTER" ?

Reply:
Ja, ser det er en relevant bekymring. Brev er jo siste steget i behandlingen (i hvert fall fra saksbehandlers perspektiv), så jeg tenker behandlingen bør låses (mulig det allerede er gjort 🤔) så snart brevet/brevrequesten lagres og behandlingen går til steget BESLUTTE_VEDTAK. Og hvis totrinns blir underkjent eller behandlingen gjøres på nytt må brevet slettes, og lages på nytt for å sikre at dataen er riktig

Reply:
Men sånn det er nå kan man jo endre behandlingen, etter man har lagret brevet, men før det er sendt til beslutter. Så da tenker jeg lagring av brevet må gjøres eksplisitt som en del av SEND_TIL_BESLUTTER.

Reply:
Hm jag hade sett for mig att man lagrer pdf:en når man totrinnskontrollen er godkjent. Fram tills dess så genererar men utkast varje gång man går in på en bruker. MEN jag har for lite kjennskap om vad konsekvenserne hvis "saksbehandler 2 ser nøyaktig samme brev som saksbehandler 1" er ikke oppnåd eller foreksempel att man gør for mange kall mot sanity